### ISSUE TITLE
Uncaught TypeError: newTarget.getComponentInChildren is not a function - Exporter 2.17.0-pre

### ISSUE THREAD

**Issue by ROBYER1** [needle-tools/needle-engine-support#46]
These scripts worked fine on earlier Exporter versions such as 2.15 and 2.16, may be related to regressions in 2.17.0-pre

I have a button in the UI calling a script that calls `getComponent` using a passed gameObject

If you pull up the repro project I sent over Wetransfer, Scenes>Test (or the only scene in there) and click the buttons in the scene you will see this behaviour too. If you roll back to 2.15.1-pre which I was using before, you will see the code works fine

Error:
```
SetCamTarget.ts:19 Uncaught TypeError: newTarget.getComponentInChildren is not a function
    at SetCamTarget.setTarget (SetCamTarget.ts:19:47)
    at invokeFunction (engine_serialization_builtin_serializer.ts:226:33)
    at CallInfo.method (engine_serialization_builtin_serializer.ts:215:57)
    at CallInfo.invoke (EventList.ts:17:14)
    at EventList.invoke (EventList.ts:31:15)
    at Button.onPointerClick (Button.ts:97:23)
    at EventSystem.handleEvents (EventSystem.ts:372:30)
    at EventSystem.handleIntersections (EventSystem.ts:259:22)
    at EventSystem.onBeforeUpdate (EventSystem.ts:180:14)
    at EventSystem.onPointerDown (EventSystem.ts:125:14)
```

**Comment by marwie** [comment]
The repro project doesnt contain your runtime code unfortunately

**Comment by ROBYER1** [comment]
Oh I have that issue I needed to report, on Windows 11 it doesn't like copying or zipping up the projects folder due to paths in it! I'll make a separate issue for that after testing more. I'll get the projects folder zipped up somehow.. 7zip complains about it

**Comment by marwie** [comment]
Paths in ``node_modules``?

Thank you

**Comment by ROBYER1** [comment]
Zip here should work, please place in projects/newProject, and yep when copying a project between locations in Windows it gets upset about paths there. I may have to always exclude node modules when moving or zipping
[newProject.zip](https://github.com/needle-tools/needle-engine-support/files/9504672/newProject.zip)

**Comment by marwie** [comment]
Do you think a button for bundling up your whole project for reporting would help / be something you'd like to use? We currently have ``Collect Logs`` but that does not copy the whole project

**Comment by ROBYER1** [comment]
> Do you think a button for bundling up your whole project for reporting would help / be something you'd like to use? We currently have `Collect Logs` but that does not copy the whole project

Yes please! We encountered issues internally with moving project folders around/zipping and always get feedback that the projects folder is missing (its a bit of a faff to sort that out when zipping or sending repros!)

**Comment by marwie** [comment]
Ok - would be great if you could send the errors you're getting (or open an issue) so we can see where exactly it's failing and provide a solution

**Comment by marwie** [comment]
I've updated to ``2.17.0-pre`` and added your project code. It looks like it works as intended here - no errors in the console unfortunately. Could you log the object ``newTarget`` before calling ``getComponentInChildren`` ?

**Comment by ROBYER1** [comment]
> Ok - would be great if you could send the errors you're getting (or open an issue) so we can see where exactly it's failing and provide a solution

For this I set up https://github.com/needle-tools/needle-engine-support/issues/47

**Comment by ROBYER1** [comment]
I have console .logged out the newTarget and it is a node object, I wonder if it doesn't report having any children. If you see the same there too let me know!

**Comment by marwie** [comment]
Yes reproduces here too, thanks for reporting

**Comment by marwie** [comment]
Fixed in 2.17.2-pre
